### PR TITLE
pg17 support

### PR DESCRIPTION
**Pull request by var77**

### CODE REVIEW

File: lantern_hnsw/scripts/test_updates.py [hunk 1]
```diff
@@ -49,7 +49,8 @@ def __repr__(self):
         return self.version
 
 INCOMPATIBLE_VERSIONS = {
-    '16': [Version('0.0.4')]
+    '16': [Version('0.0.4')],
+    '17': [Version('0.3.0'), Version('0.3.1'), Version('0.3.2'), Version('0.3.3'), Version('0.3.4'), Version('0.4.0'), Version('0.4.1')],
```
Comment (Ngalstyan4) [inline lantern_hnsw/scripts/test_updates.py:53]: If we no longer have servers at an older `0.3.*` version that we need to upgrade from, you can also move old update scripts to [old_updates](https://github.com/lanterndata/lantern/tree/main/lantern_hnsw/sql/old_updates)

File: lantern_hnsw/test/expected/hnsw_vector.out [hunk 2]
```diff
@@ -199,48 +184,43 @@ CREATE INDEX l2_idx ON small_world_arr USING lantern_hnsw(v) WITH (dim=3, m=2);
...
+SELECT id FROM small_world_arr ORDER BY v <=> ARRAY[0,0,0];
  id 
 ----
   1
   2
   3
-(3 rows)
 
 DROP INDEX cos_idx;
 CREATE INDEX ham_idx ON small_world_arr USING lantern_hnsw(v) WITH (m=3);
```
Comment (Ngalstyan4) [inline lantern_hnsw/test/expected/hnsw_vector.out:213]: was this supposed to be testing hamming distance? it seems the index creation uses the default l2sq operator in this case as well - there is a typo in the old test.

**Review by Ngalstyan4 — APPROVED** [review]

File: lantern_hnsw/test/expected/hnsw_vector.out (hunk 2 above)
Comment (var77) [reply lantern_hnsw/test/expected/hnsw_vector.out]: fixed to create an index with correct opclass

File: lantern_hnsw/scripts/test_updates.py (hunk 1 above)
Comment (var77) [reply lantern_hnsw/scripts/test_updates.py:53]: there are still servers with 0.3.x versions, will move the update files to old_updates once all servers are upgraded

**Pull request merged**